pkg/registries/docker/metadata_v2.go

```diff
@@ -19,7 +20,12 @@ func HandleV2ManifestList(r *Registry, remote, ref string) (*storage.ImageMetada
 		// Default to linux arch
 		// TODO(ROX-13284): Support multi-arch images.
 		if manifest.Platform.OS == "linux" && manifest.Platform.Architecture == "amd64" {
-			return HandleV2Manifest(r, remote, manifest.Digest.String())
+			switch manifest.Descriptor.MediaType {
```

Comment:
This is fine, but isn't the image in question in the other direction? It's an OCI Image Index with a reference to an amd64 Docker v2 Manifest? Should we add this same logic to `HandleOCIImageIndex`?